openstates/oh/bills.py

```diff
@@ -13,43 +13,333 @@
+                            "msg_reso_503": "bill:passed",
+                            "intro_107": ["bill:introduced","bill:passed"],
+                            "imm_consid_360": "bill:passed"}
+
+            
+
+            base_url = "http://search-prod.lis.state.oh.us/"
+            first_page = base_url + "/solarapi/v1/general_assembly_{session}/".format(session=session)
+            legislators = self.get_legislator_ids(first_page)
+            all_amendments = self.get_other_data_source(first_page,base_url,"amendments")
+            all_fiscals = self.get_other_data_source(first_page,base_url,"fiscals")
+            all_synopsis = self.get_other_data_source(first_page,base_url,"synopsiss")
+            all_analysis = self.get_other_data_source(first_page,base_url,"analysiss")
+            doc_types = ["bills","resolutions"]
+            for doc_type in doc_types:
+                bill_versions = {}
+                for doc in self.pages(base_url,first_page+doc_type):
+                    for v in doc["items"]:
+                        #bills is actually a list of versions
+                        #going to create a dictionary as follows:
+                        #key=bill_id
+                        #value = dict of all versions, where keys are versionids
+                        #and values are the bill data from the api.
+                        #then loop through that to avoid duplicate bills
+
+                        bill_id = v["number"]
+                        version_id = v["versionid"]
+                        if bill_id in bill_versions:
+                            if version_id in bill_versions[bill_id]:
+                                self.logger.warning("There are two versions of {bill_id} called the same thing. Bad news bears!".format(bill_id=bill_id))
+                            else:
+                                bill_versions[bill_id][version_id] = v
+                        else:
+                            bill_versions[bill_id] = {}
+                            bill_versions[bill_id][version_id] = v
+
+                    
+
+                for b in bill_versions:
+                    bill = None
+                    for bill_version in bill_versions[b].values():
+                        if not bill:
+                            bill_id = bill_version["number"]
+                            title = bill_version["shorttitle"] or bill_version["longtitle"]
+
+                            title = title.strip()
+
+                            chamber = chamber_dict[bill_version["chamber"]]
+
+                            subjects = []
+                            for subj in bill_version["subjectindexes"]:
+                                try:
+                                    subjects.append(subj["primary"])
+                                except KeyError:
+                                    pass
+
+
+                            bill = Bill(session,chamber,bill_id,title,subjects=subjects,type=doc_type)
```

Comment:
`bill_id` (at least in this instantiation argument) should be uppercase, and have a space between the letters and numbers. This still seems to be their convention, despite their LIS's API; [see a search page's return on their main website](https://www.legislature.ohio.gov/legislation?28&pageSize=10&start=1&sort=LegislationNumber&dir=asc&statusCode&legislationTypes=HB&generalAssemblies=131).